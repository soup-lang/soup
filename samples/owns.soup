class Point
    ctor(x, y)
        this.x: x
        this.y: y

    func toString()
        return '(' + this.x + ', ' + this.y + ')'

p: Point(3.14, 2.7)
console.log(p.toString())
console.log(p owns 'x')
console.log(p owns 'y')
console.log(p owns 'toString')
console.log(p.toString != undefined)
console.log(p.__other__ != undefined)

console.log('+ list index')
pl: [Point(0, 0), Point(0, 1), Point(1, 0)]
for i range 4
    console.log(pl owns i)

console.log('+ delete x')
delete p.x
console.log(p owns 'x')
console.log(p owns 'y')
console.log(p owns 'toString')
console.log(p.toString != undefined)
console.log('+ delete toString')
delete p.toString
console.log(p owns 'x')
console.log(p owns 'y')
console.log(p owns 'toString')
console.log(p.toString != undefined)
