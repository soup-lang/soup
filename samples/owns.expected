(3.14, 2.7)
true
true
false
true
false
+ list index
true
true
true
false
+ delete x
false
true
false
true
+ delete toString
false
true
false
true
